### DISCUSSION TITLE
how do we pass around variables?

### DISCUSSION THREAD

**Discussion by tsvetov11** [postmanlabs/postman-flows#382] [Defect]
hey, I am using flows to test multiple APIs in a sequence, here's where I run into a problem

the first API sends out some data, such as phone model, app version, language, etc.

the second API is signup, thus we are using random variables for the email and password - {{$randomFirstName}}@example.com and {{$randomPassword}}, and additionally, I have written scripts that take the email and password from the request data and set it as a variable.

the third API is a login, using the variables mentioned above

when running the collection/folder for these APIs, they all work fine, all of the variables are set and used properly in the later APIs

after creating a flow, with the APIs, I notice that they do not set/update variables, and I cannot seem to find a way to pass this data around as durables.
Additionally, not all of the console.log scripts in each API get executed  and printed when running them in a flow for some reason (for instance, in the signup request I've got 5 console.log that all execute and print properly when running the request but itself or in the runner, but using the flows, I only see one of them executing and printing)

can anyone actually explain why only some of the scripts per request are running?
any help is appreciated!

**Comment by Ariemeth** [comment]
@tsvetov11 I'm not sure that will work with flows as it currently is working.
To my understanding, pre and test scripts cannot modify environment values when running as part of a flow.  You also cannot set durables or data to be something from an environment value.  Environment values can only be used by a `Send Request` block.  

I've gotten around this issue by generating usernames as data in the and use a set password that is stored in the flow.

As for the console.logs I do not know.

**Reply by Ariemeth** [reply]
Here is an example of what I'm doing
![image](https://user-images.githubusercontent.com/541023/186476413-54a34008-bc00-4514-a934-28f5c61f5eac.png)

If your API that handles the creation of an account returns the username you could scrape that from the response.  The final `Create Data` that is visible in the screenshot does that.  It grabs the user's system generated ID and username from the response from `Create Test User`

**Reply by tsvetov11** [reply]
how exactly can i set a durable or data block to get something from an environment variable's value?

as mentioned in my post, I'm not using usernames, I'm using email/pass combo, and no, my API doesn't return any of those in the response

**Reply by Ariemeth** [reply]
I do not believe you can do that currently
